### PR TITLE
AG-921 add biodomains filterset

### PR DESCRIPTION
**Pull request by sagely1**

### CODE REVIEW

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts [hunk 1]
```diff
@@ -211,18 +209,17 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
     this.helperService.setLoading(true);
     this.genes = [];
 
-    this.geneService
-      .getComparisonGenes(this.category, this.subCategory)
-      .subscribe((res: GCTGeneResponse) => {
-        this.initData(res.items);
+    const genesApi$ = this.geneService.getComparisonGenes(this.category, this.subCategory);
+    const distributionApi$ = this.geneService.getDistribution();
+
```
Comment (sagely1) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts:214]: using RxJS to call both apis.

File: src/app/features/genes/services/gene.service.ts [hunk 2]
```diff
@@ -141,6 +146,30 @@ export class GeneService {
 
   // ------------------------------------------------------------------------ //
```
Comment (sagely1) [inline src/app/features/genes/services/gene.service.ts:148]: new api methods

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.variables.ts [hunk 3]
```diff
@@ -97,6 +97,92 @@ export const filters: GCTFilter[] = [
       },
     ],
   },
```
Comment (sagely1) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.variables.ts:99]: new biodomain filters

File: src/server/api.ts [hunk 4]
```diff
@@ -84,6 +86,8 @@ connection.on(
 const router = express.Router();
 
 connection.once('open', async () => {
```
Comment (sagely1) [inline src/server/api.ts:88]: new api routes

File: src/server/components/comparison.ts [hunk 5]
```diff
@@ -112,8 +112,9 @@ function getComparisonGeneNominations(gene: Gene, teams: Team[]) {
   return data;
 }
 
-function getComparisonGene(gene: Gene, teams: Team[], scores: Scores[]) {
+function getComparisonGene(gene: Gene, teams: Team[], scores: Scores[], allBiodomains: BioDomains[] | undefined) {
   const geneScores = scores.find(score => score.ensembl_gene_id == gene.ensembl_gene_id);
+  const geneBiodomains = allBiodomains?.find(b => b.ensembl_gene_id === gene.ensembl_gene_id)?.gene_biodomains.map(b => b.biodomain);
```
Comment (sagely1) [inline src/server/components/comparison.ts:118]: This is technical debt from before my time.  Ideally we would call the gene collection and wouldn't have to join the biodomains / score data as this is not taking advantage of MongoDB.

**Review by hallieswan — APPROVED** [review]
Looks good!

**Pull request merged**